### ISSUE TITLE
call termopen in loop throw error: uv__io_poll: Assertion `(u nsigned) fd < loop->nwatchers'

### ISSUE THREAD

**Issue by kevinhwang91** [neovim/neovim#13874]
- `nvim --version`: v0.5.0-dev+1062-gcc1851c9f
- `vim -u DEFAULTS` (version: ) behaves differently?
- Operating system/version: ArchLinux
- Terminal name/version: Alacritty
- `$TERM`: tmux-256color

### Steps to reproduce using `nvim -u NORC`

```
1. nvim -u NORC +"call termopen('ls')"
2. nvim -u NORC -c "autocmd VimEnter * call termopen('ls')"
3. nvim -u NORC +"call timer_start(1000, {->call('termopen', ['ls', {}])})"
```

### Actual behaviour
1. open terminal and list files for current work directory.
2. throw below error
3. throw below error after 1s

```log
nvim: xxx/neovim/.deps/build/src/libuv/src/unix/linux-core.c:384: uv__io_poll: Assertion `(unsigned) fd < loop->nwatchers' failed.

[Process exited 134]
```

### Expected behaviour
case 2 and 3 work as expected.

I bisected from the log and found https://github.com/neovim/neovim/pull/12937 break the change.

**Comment by jamessan** [comment]
An internal error is happening in the child process before we exec the new process.  That internal error is causing a "hit enter" prompt, but since it's in the child there's no way to address it.

Seems like there are two issues here.

* After we fork, nvim shouldn't think it can still interact with the user.
* Figure out why in scenarios 2 and 3 there's already a `TERM` entry in the env dict in `init_child()` and either fix that or override the value

**Comment by jamessan** [comment]
>     * Figure out why in scenarios 2 and 3 there's already a `TERM` entry in the env dict in `init_child()` and either fix that or override the value

Ah, before we supported specifying the environment for a pty job, we used the `TERM` key from `jobstart()`'s options dict to set it.  Now, the environment is inheriting nvim's `$TERM` from the environment automatically, so when we try to explicitly set it in the child we hit the internal error for adding a key to a dict when one already exists.

We should probably deprecate that in favor of using `env`.
